pkg/controller/operators/olm/operator.go

```diff
@@ -1415,27 +1416,13 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 			return
 		}
 
-		// Create APIService
-		for _, desc := range out.GetOwnedAPIServiceDescriptions() {
-			if deploymentCAPEMs == nil {
-				err = fmt.Errorf("Deployment CAPEM map should not be nil")
-				return
-			}
-			caPEM, ok := deploymentCAPEMs[desc.DeploymentName]
-			if !ok {
-				err = fmt.Errorf("Deployment not associated with APIService")
-				return
-			}
-			err = a.createOrUpdateAPIService(caPEM, desc, out)
-			if err != nil {
-				return
-			}
-
-			// Cleanup legacy resources
-			err = a.deleteLegacyAPIServiceResources(out, desc)
-			if err != nil {
-				return
-			}
+		if len(out.GetAllAPIServiceDescriptions()) > 0 {
+			expiration := time.Now().Add(DefaultCertValidFor)
```

Comment:
For consistent start times:
```suggestion
			now := metav1.Now()
			expiration := now.DeepCopy().Add(DefaultCertValidFor)
```

Reply:
Good idea!

Reply:
Addressed in the latest version of the PR.